### PR TITLE
Observability in library and statime linux

### PR DESCRIPTION
**Pull request by cikzh**
This PR will add a Prometheus exporter, which will get its metrics from statime-linux via a Unix Domain Socket, very similar to how ntpd-rs does it.

It implements all currently available Instance datasets and exposes only a small set of metrics. The full exposure of metrics will be implemented in a separate PR

### CODE REVIEW

File: statime/src/ptp_instance.rs [hunk 1]
```diff
@@ -161,6 +165,26 @@ impl<F> PtpInstance<F> {
             _filter: PhantomData,
         }
     }
+
+    /// Return IEEE-1588 defaultDS for introspection
+    pub fn default_ds(&self) -> ObservableDefaultDS {
+        self.state.borrow().default_ds.into()
```
Comment (rnijveld) [inline statime/src/ptp_instance.rs:171]: DefaultDS is copied here implicitly and then copied once more with the into (I'm not sure why DefaultDS and CurrentDS implement clone, seems like a papercut).

File: statime/src/ptp_instance.rs [hunk 2]
```diff
@@ -161,6 +165,26 @@ impl<F> PtpInstance<F> {
...
+    }
+
+    /// Return IEEE-1588 currentDS for introspection
+    pub fn current_ds(&self) -> ObservableCurrentDS {
+        self.state.borrow().current_ds.into()
+    }
+
+    /// Return IEEE-1588 parentDS for introspection
+    pub fn parent_ds(&self) -> ObservableParentDS {
+        self.state.borrow().parent_ds.clone().into()
```
Comment (rnijveld) [inline statime/src/ptp_instance.rs:181]: This clone could then be removed.

File: statime/src/ptp_instance.rs [hunk 3]
```diff
@@ -161,6 +165,26 @@ impl<F> PtpInstance<F> {
...
+    }
+
+    /// Return IEEE-1588 timePropertiesDS for introspection
+    pub fn time_properties_ds(&self) -> ObservableTimePropertiesDS {
+        self.state.borrow().time_properties_ds.into()
```
Comment (rnijveld) [inline statime/src/ptp_instance.rs:186]: Same for this one.

**Review by rnijveld — CHANGES_REQUESTED** [review]

**Review by davidv1992 — APPROVED** [review]
Note: Reverts #405 but I am fine with that as that did not fix the issue anyway.

**Review by davidv1992 — CHANGES_REQUESTED** [review]
Apologies, missed that sdoId still has the default serde deserializer, which is problematic as an sdo id should be below 2^12, which this doesn't enforce.

**Review by davidv1992 — APPROVED** [review]

File: statime/src/datastructures/datasets/time_properties.rs [hunk 4]
```diff
@@ -7,13 +7,22 @@ use crate::datastructures::common::{LeapIndicator, TimeSource};
 ///
 /// For more details see *IEEE1588-2019 section 8.2.4*.
 #[derive(Default, Copy, Clone, Debug, Eq, PartialEq)]
```
[the remark this answers is not in the document]
Comment (davidv1992) [reply statime/src/datastructures/datasets/time_properties.rs:9]: I disagree, this is small plain old data, so I think it should be copy.

**Review by rnijveld — APPROVED** [review]

**Pull request merged**